### PR TITLE
YARN-11459 Changed label called "max resource" on UIv1 and UIv2

### PR DESCRIPTION
**Pull request by riyakhdl**
[YARN-11459](https://issues.apache.org/jira/browse/YARN-11459)

Consider changing label called "max resource" on UIv1 and UIv2

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 35s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 55s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  25m 53s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   9m 39s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  compile  |   8m 27s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  checkstyle  |   1m 43s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 57s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m 19s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   1m  8s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +0 :ok: |  spotbugs  |   0m 25s |  |  branch/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-ui no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  19m 32s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 27s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m  9s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   9m  2s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javac  |   9m  2s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 25s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  javac  |   8m 25s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m 37s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 45s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m  2s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   0m 59s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +0 :ok: |  spotbugs  |   0m 19s |  |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-ui has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  19m 28s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  98m 30s |  |  hadoop-yarn-server-resourcemanager in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 29s |  |  hadoop-yarn-ui in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 56s |  |  The patch does not generate ASF License warnings.  |
|  |   | 241m 39s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5527/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5527 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 9ef843cbbf1e 4.15.0-206-generic #217-Ubuntu SMP Fri Feb 3 19:10:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / fa0f7f4860d97a29fb11b232817724fb1545c094 |
| Default Java | Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5527/1/testReport/ |
| Max. process+thread count | 920 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager hadoop-yarn-project/hadoop-yarn/hadoop-yarn-ui U: hadoop-yarn-project/hadoop-yarn |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5527/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Review by susheelgupta7 — APPROVED** [review]

**Review by hotcodemacha — CHANGES_REQUESTED** [review]
Thanks @riyakhdl for your PR. Can you share the motivation/requirement behind this change ?

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-ui/src/main/webapp/app/templates/yarn-queue/apps.hbs [hunk 1]
```diff
@@ -50,7 +50,7 @@
           <thead>
             <tr>
               <th>User Name</th>
-              <th>Max Resource</th>
+              <th>User Limit Resource</th>
```
Comment (ayushtkn) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-ui/src/main/webapp/app/templates/yarn-queue/apps.hbs:53]: ```User Resource Limit``` sounds better to me

**Review by ayushtkn — COMMENTED** [review]
I am also curious what is the motivation behind this change. Should do only if we have a good reason, else people used to the old name would get confused what is this new thing

**Comment by riyakhdl** [comment]
Thanks @ashutoshcipher & @ayushtkn for reviewing, 
This change was proposed during an escalation, where the customer had confusion about the difference between "Max Resource" and "Effective Max Capacity" for a dedicated queue.  
Here is the doc explaining how “Max Resource” is calculated.   Thanks to Szilard Nemeth for the detailed explanation.
[Max resource calculation.zip](https://github.com/apache/hadoop/files/11307793/Max.resource.calculation.zip)

So as seen from the calculation "Max Resource" is actually from userLimitResource, hence it seems better to change the label “Max Resource” to “User Limit Resource".

**Comment by ayushtkn** [comment]
Thanx @riyakhdl, makes sense to me.

**Review by hotcodemacha — APPROVED** [review]

**Comment by brumi1024** [comment]
Thanks @riyakhdl for the patch, and @ashutoshcipher @ayushtkn @susheelgupta7 for the review. Merged to trunk.

**Pull request merged**